### PR TITLE
[Implementation][EFIP-4] Process Fees in EETH

### PR DESCRIPTION
**Pull request by vvalecha519**
https://hackmd.io/@sK3pnSmVSxekZ0_510u48g/rkBMangsA

### CODE REVIEW

File: src/LiquidityPool.sol [hunk 1]
```diff
@@ -461,6 +462,20 @@ contract LiquidityPool is Initializable, OwnableUpgradeable, UUPSUpgradeable, IL
 
         emit Rebase(getTotalPooledEther(), eETH.totalShares());
     }
+    /// @notice pay protocol fees including 5% to treaury, 5% to node operator and ethfund bnft holders
+    /// @param _protocolFees The amount of protocol fees to pay in ether
+    function payProtocolFees(uint128 _protocolFees) external {
+        if (msg.sender != address(etherFiAdminContract)) revert IncorrectCaller();
```
Comment (seongyun-ko) [inline src/LiquidityPool.sol:468]: Is it necessary though the caller is being checekd on `depositToRecipient` function

**Comment by seongyun-ko** [comment]
- could u add EFIP-4 proposal to it too? https://github.com/etherfi-protocol/smart-contracts/blob/staging-2.5/proposals/EFIP-4.md
- could u update EFIP-4 proposal regarding the one-time big payment processing? we will need to comm this or get prepared at least

File: src/LiquidityPool.sol (hunk 1 above)
Comment (vvalecha519) [reply src/LiquidityPool.sol:468]: not necessary just more clear and its a little stronger because if I remove it liquifier contract can call method.

**Review by seongyun-ko — APPROVED** [review]
lgtm

**Comment by seongyun-ko** [comment]
can u also update EFIP-4 with the audit reviews from Nethermind (link its reference) and Certora (upload the draft review and link the reference)

**Pull request merged**